### PR TITLE
fix(upgrade_test): forget the scylla_version after node upgrade

### PR DESCRIPTION
**Pull request by fruch**
we recently introduced code that does logic based on `node.scylla_version` at the end of the upgrade sequence, and it wasn't working as we expected, returning the base version.

we now in the upgrade sequence clear the cached for scylla_version related functions/attributes

Ref: https://github.com/scylladb/scylladb/issues/16147

### Testing
- [x] https://jenkins.scylladb.com/job/scylla-staging/job/fruch/job/rolling-upgrade-debian11-test/7/
- [x] https://jenkins.scylladb.com/job/scylla-staging/job/fruch/job/rolling-upgrade-ubuntu20.04-test/16/

## PR pre-checks (self review)

- [ ] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [ ] I didn't leave commented-out/debugging code
- [ ] I added the relevant `backport` labels
- [ ] New configuration option are added and documented (in `sdcm/sct_config.py`)
- [ ] I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)
- [ ] All new and existing unit tests passed (CI)
- [ ] I have updated the Readme/doc folder accordingly (if needed)

### CODE REVIEW

**Review by tchaikov — APPROVED** [review]
i approve this change. because i want to get this fix landed sooner.

i still prefer https://github.com/scylladb/scylla-cluster-tests/pull/6846, because

1. it's created first and it follows the work flow i'd recommend =)
2. it has better commit message

but please, in future, please self-assign a ticket and coordinate with the existing assignee(s). so we can avoid the overhead.

**Comment by fruch** [comment]
> 
> but please, in future, please self-assign a ticket and coordinate with the existing assignee(s). so we can avoid the overhead.

you self assigned it, while I was looking into it, regardless it's a regression cause by a code I introduced this week.

**Comment by fruch** [comment]
> i approve this change. because i want to get this fix landed sooner.

It would land after it would be tested, ti would take a few hours. 

> 
> i still prefer #6846, because
> 
> 1. it's created first and it follows the work flow i'd recommend =)
> 2. it has better commit message
> 
> but please, in future, please self-assign a ticket and coordinate with the existing assignee(s). so we can avoid the overhead.

**Comment by tchaikov** [comment]
> > i approve this change. because i want to get this fix landed sooner.
> 
> It would land after it would be tested, ti would take a few hours.
> 

i mean, if i don't get the approval. i am willing to review another change.

**Comment by tchaikov** [comment]
> > but please, in future, please self-assign a ticket and coordinate with the existing assignee(s). so we can avoid the overhead.
> 
> you self assigned it, while I was looking into it, regardless it's a regression cause by a code I introduced this week.

right. because i am willing to fix it.

i still suggest we follow the convention. anything i am missing?

**Comment by mykaul** [comment]
@vponomaryov , @soyacz - please review promptly - we need to get it in (and backport to 5.4) to unlock 5.4 upgrade tests.

**Comment by vponomaryov** [comment]
> @vponomaryov , @soyacz - please review promptly - we need to get it in (and backport to 5.4) to unlock 5.4 upgrade tests.

Why not https://github.com/scylladb/scylla-cluster-tests/pull/6846 ?

**Review by vponomaryov — APPROVED** [review]

**Comment by fruch** [comment]
> > @vponomaryov , @soyacz - please review promptly - we need to get it in (and backport to 5.4) to unlock 5.4 upgrade tests.
> 
> Why not #6846 ?

code wise they are the same, I'll merge and backport once test were passing

**Comment by fruch** [comment]
> > > @vponomaryov , @soyacz - please review promptly - we need to get it in (and backport to 5.4) to unlock 5.4 upgrade tests.
> > 
> > 
> > Why not #6846 ?
> 
> code wise they are the same, I'll merge and backport once test were passing

and I see they just did

**Pull request merged**
